zetasql/fuzzing/protobuf/zetasql_expression_extractor_test.cc

```diff
@@ -0,0 +1,267 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <memory>
+#include <string>
+
+#include "zetasql/fuzzing/protobuf/zetasql_expression_grammar.pb.h"
+#include "zetasql/fuzzing/protobuf/zetasql_expression_extractor.h"
+#include "gtest/gtest.h"
+
+using zetasql_expression_grammar::Expression;
+using zetasql_expression_grammar::LiteralExpr;
+using zetasql_expression_grammar::IntegerLiteral;
+using zetasql_expression_grammar::NumericLiteral;
+using zetasql_expression_grammar::CompoundExpr;
+using zetasql_expression_grammar::BinaryOperation;
+using parameter_grammar::Whitespace;
+using zetasql_fuzzer::ProtoExprExtractor;
+
+namespace zetasql_fuzzer {
+
+class ProtoExprExtractorTest : public ::testing::Test {};
+
+TEST_F(ProtoExprExtractorTest, ExtractorReleaseTest) {
+  std::string arbitrary("asdAgwegfGgw11");
+  ProtoExprExtractor extractor;
+
+  EXPECT_EQ(extractor.Release(), "");
+  extractor.Append(arbitrary);
+  EXPECT_EQ(arbitrary, "asdAgwegfGgw11");
+  EXPECT_EQ(extractor.Release(), "asdAgwegfGgw11");
+  EXPECT_EQ(extractor.Release(), "");
+}
+
+namespace {
+
+TEST_F(ProtoExprExtractorTest, UninitializedOneOfExprTest) {
+  ProtoExprExtractor extractor;
+
+  Expression expr;
+  extractor.Extract(expr);
+  EXPECT_EQ(extractor.Release(), "");
+
+  expr.mutable_default_value()->set_content("default");
+  extractor.Extract(expr);
+  EXPECT_EQ(extractor.Release(), "default");
+
+  LiteralExpr lit_expr;
+  extractor.Extract(lit_expr);
+  EXPECT_EQ(extractor.Release(), "");
+
+  lit_expr.mutable_default_value()->set_content("default_lit");
+  extractor.Extract(lit_expr);
+  EXPECT_EQ(extractor.Release(), "default_lit");
+
+  IntegerLiteral int_expr;
+  extractor.Extract(int_expr);
+  EXPECT_EQ(extractor.Release(), "");
+
+  int_expr.mutable_default_value()->set_content("default_int");
+  extractor.Extract(int_expr);
+  EXPECT_EQ(extractor.Release(), "default_int");
+
+  CompoundExpr comp_expr;
+  extractor.Extract(comp_expr);
+  EXPECT_EQ(extractor.Release(), "");
+
+  comp_expr.mutable_default_value()->set_content("default_expr");
+  extractor.Extract(comp_expr);
+  EXPECT_EQ(extractor.Release(), "default_expr");
+}
+
+TEST_F(ProtoExprExtractorTest, SpecialLiteralTest) {
+  using SpecialVal = zetasql_expression_grammar::LiteralExpr_SpecialValue;
+  LiteralExpr lit_expr;
+  ProtoExprExtractor extractor;
+  lit_expr.set_special_literal(SpecialVal::LiteralExpr_SpecialValue_V_NULL);
+  extractor.Extract(lit_expr);
+  EXPECT_EQ(extractor.Release(), "NULL");
+}
+
+TEST_F(ProtoExprExtractorTest, BytesLiteralTest) {
```

Comment:
This is three unit tests.

Reply:
Per conversation yesterday:  Maybe use TEST_P here.